### ISSUE TITLE
Extract more metadata: flag “is informative”

### ISSUE THREAD

**Issue by tripu** [w3c/specberus#143]
Issue #106 spun off this one, post-v1.
Some ideas (moved here from the original):

* ~~Patent Policy maturity (always 'None'; 0)~~
* ~~Patent Policy status (URL from the document, or *Not in Rec Track* if informative (given in input, therefore from registration)~~
* Is Informative
* ~~Editor's draft~~

**Comment by astorije** [comment]
- @tripu, IMO you can remove `Patent Policy maturity` and `Patent Policy status`. These need to be handled by our patent policy system, not by Specberus, not by Echidna, not by the publication system at W3C.
- Informative, I'll build examples
- Editor's draft is no worky? We are using it on [Echidna](https://github.com/w3c/echidna/blob/master/lib/publisher.js#L42). This blocks v1 IMO.

**Comment by tripu** [comment]
@astorije:
Description of the issue updated as per your comments.
As for the editor's draft, the parent issue was about *metadata*, so I suspect this refers (or should refer) to Echidna *returning* info to say whether the original document was an editor's draft&hellip; no?

**Comment by astorije** [comment]
Hum, that's a good point. Are EDs a different profile?

If so, we don't need something like `isEditorsDraft` as it would be duplicate of the profile. On the other hand, I don't know if EDs specify if they are editor's drafts of WDs, RECs, ...

As far as I am concerned right now, I just need to retrieve the Editor's Draft URI. For the rest, I'm not sure. Maybe having other's opinions might help?

**Comment by tripu** [comment]
So, what about:

* No changes to profiles.
* Add an `editorsDraftURL` key to the metadata returned by Echidna. If the document was not an editor's draft, it simply won't be returned.

At first sight, it seems easy to pass along the original URL found in the manifest file (which is the editor's draft URL) and return it at the end of the process from Echidna.

**Comment by astorije** [comment]
Hey @tripu, I actually wonder if we are talking about the same thing here. The editor's draft I am only talking about is the URL that can be found in the header, usually under "Latest editor's draft", like here: http://www.w3.org/TR/2015/WD-mediacapture-streams-20150212/.

These links can these be found on TR under "Nightly Draft" links, like here: http://www.w3.org/TR/tr-groups-all#tr_Device_APIs_Working_Group.

Does that make more sense?

**Comment by tripu** [comment]
@astorije: D'oh! My bad&hellip;

Of course [we're returning that piece of metadata](https://github.com/w3c/specberus/commit/af95f4985add7c70656f8bf69900302aa5ee07a6). I was confused because I couldn't find any mention to that in the `README`. It turns out [I removed the documentation about that by accident](https://github.com/w3c/specberus/commit/ca435e3d7708660a30170faef8b087ea43d1bafd#diff-04c6e90faac2675aa89e2176d2eec7d8L92) :(

[It's fixed now in a branch](https://github.com/w3c/specberus/commit/043936cd93d3418fc81acdb2f95297db20594bb7), and it will be part of a new PR soon. So I'm crossing that one out too, from the description of this issue. Thanks!

**Comment by tripu** [comment]
@astorije, sorry, where shall I look in a document to know whether it's informative, or not?

Those examples you mentioned&nbsp;&mdash;&nbsp;shall we edit the sample documents already used by the test suite, to make some of them *informative*?

**Comment by astorije** [comment]
@tripu, here are a few documents to show examples of what I meant weeks ago:

1. [Metadata Vocabulary for Tabular Data](http://www.w3.org/TR/2015/WD-tabular-metadata-20150108/)
This is an **explicit normative** document because you can read in the SOTD: *This document is intended to become a W3C Recommendation.*

2. [CSS Exclusions Module Level 1](http://www.w3.org/TR/2015/WD-css3-exclusions-20150115/)
This is an **implicit normative** document: if nothing is specified, then we consider it normative.

3. [MediaStream Recording](http://www.w3.org/TR/2015/WD-mediastream-recording-20150127/)
This is an **explicit informative** document, as this can be read on the patent policy paragraph: *The group does not expect this document to become a W3C Recommendation.*

4. [Predefined Counter Styles](http://www.w3.org/TR/2015/WD-predefined-counter-styles-20150203/)
This one is an **explicit informative** document as this can be read in the PP paragraph: *The group does not expect this document to become a W3C Recommendation.*
Moreover, this can be found in the SOTD, to reflect case 1: *The Working Group expects this document to become a Working Group Note.*

5. [HTML5 Differences from HTML4](http://www.w3.org/TR/2014/WD-html5-diff-20140918/)
This was **incorrectly detected** as a normative document. Although *The Working Group intends to publish this document as a Working Group Note.* was specified in the SOTD, nothing was written in the patent policy paragraph, so the detection defaulted to normative.

6. [Character Model for the World Wide Web: String Matching and Searching](http://www.w3.org/TR/2014/WD-charmod-norm-20140715/)
Same as case 5 with different wording: "This document is intended to become a W3C Working Group Note."

7. [Using WAI-ARIA in HTML](http://www.w3.org/TR/2014/WD-aria-in-html-20140626/)
This last case IMO is **ambiguous**. It doesn't say if it should be normative or informative, so it is detected as normative, but most often "guidelines" documents are actually informative (Pubrules even puts a warning when the title contains words such as "Guidelines" or "Practices").

In the end, cases that must be avoided (5 and 6) and cases that should be avoided (7, maybe others I didn't cover) only come from implicit detection (case 2, and case 1 as well actually, if detection is only done on the PP paragraph).

I think we really want one place, the PP section for example, to have one statement with 2 options: normative or informative. The wording can be same as case 1 (for normative) and case 3 (for informative), but defaulting a normative results in wrong detection, many times. Yes it's one more requirement for editors (although not really, they almost all specify it, with different wording each time, I am just suggesting a standard but mandatory statement), but that would be for the best.

Would @dontcallmedom and @darobin, WG-involved people, be bothered by that?

**Comment by dontcallmedom** [comment]
Wouldn't it make more sense to set at FPWD time (in the registration system)  whether a document is expected to be normative or not?

Note that to be precise, a document can be non-normative yet target to become a  W3C Recommendation — these aren't mutually exclusive (the normativeness impacts how the document is handled from a patent policy perspective thou

**Comment by darobin** [comment]
Note that this can change over time, too, as groups update their plans.

**Comment by astorije** [comment]
@dontcallmedom, the registration system is here for authentication only and I hope it will stay that way unless it becomes mandatory. The more we rely on data coming from it, the more spaghetti we'll have between our systems. It will be harder to debug and to make it evolve. That way, we will be able to use Echidna for FPWDs as well (you register prior to FPWD) and we only rely on Specberus for data extraction.
The other main reason is that, as @darobin said, things can change over time and you'll need someone to update what was registered before, which seems odd to me.
In general, I think exclusively relying on what is actually in a document is a good idea and we'll benefit from it in the future, as it removes inconsistencies, improves transparency, ... I also think we can have that without any overhead from the editors' side. Keep in mind that making their lives easier is my (almost) only goal :-)

That said, I actually didn't know that a document can be non-normative **and** be expected to become a W3C Recommandation... But in this case, then I have no way to tell the systems today and if this happened, then we have inconsistencies in the database for sure.

So to summarize, the sentence that says either "expected to become a Rec" or "expected to become a Note" is irrelevant, right?
Anyway, my comment still applies, I think relying on default values/texts to detect data is a bit tricky as forgetting to add this text outputs the wrong metadata without being able to raise an error. Am I being clear here?

**Comment by dontcallmedom** [comment]
You're right that mixing up the registration system here is probably at least premature, and maybe just wrong. 

That being said, I don't think pubrules should (nor could) be only about the self-contained document — a major role of pubrules is to ensure a document is consistent with the world in which it is published (from style, IPR, historical, etc perpsectives).

It may be that relying on specific text or hook in the document will work best, but I think that can only be assessed from a workflow perspective (will an editor think of making the change if there is a change in normativity? how can we detect when such a change happen? should we think of out-of-band processes for recording when such a change happen?)

Re "expected to become a Rec" vs "a Note", it's not irrelevant: Note clearly implies informative; Rec is ambiguous (although it would be probably fair to say that unless stated otherwise, a Rec is likely a normative document).

In any case, I do agree with you that there seems to be ambiguity both in rules and usage in this space that's probably worth digging into; I would just warn about thinking of this issue as a pure "pubrules" issue — it should be thought in the broader context of W3C process and publication workflows in groups.

**Comment by astorije** [comment]
> That being said, I don't think pubrules should (nor could) be only about the self-contained document — a major role of pubrules is to ensure a document is consistent with the world in which it is published (from style, IPR, historical, etc perpsectives).

Agreed, and we can have Specberus interact with interfaces from this world. For example, at some point the Echidna profiles will check that the editor IDs actually exist, and for that they'll have to discuss with our DB...

> In any case, I do agree with you that there seems to be ambiguity both in rules and usage in this space that's probably worth digging into; I would just warn about thinking of this issue as a pure "pubrules" issue — it should be thought in the broader context of W3C process and publication workflows in groups.

I understand your warning and ack it :)
I simply wanted to report that we need something more to clearly identify this on the document, as there are known false negatives on the subject (cases 5, 6 and to some extent 7).

Thanks for your input!

**Comment by astorije** [comment]
I now realize I had misunderstood the informative bit. I just published a FPWD from an Interest Group, intended to become a W3C REC. However, it was published as *Informative*, as explained by @ianbjacobs:

> &lt;Ian&gt; Informative only is ok
> &lt;Ian&gt; Here's why:
> &lt;Ian&gt; We basically say that only normative Recs from WGs are normative.
> &lt;Ian&gt; Everything else is informative:
> &lt;Ian&gt;  * Recs from WGs that are informative only
> &lt;Ian&gt;  * Recs from IGs of any kind
> &lt;Ian&gt;  * Notes
> &lt;Ian&gt; Normative here means "Software has to do this and we want patent commitments over it."
> &lt;Ian&gt; examples:
> &lt;Ian&gt;  * A primer (which just explains something) is not meant to be implemented under RF licensing terms.
> &lt;Ian&gt; So it is "informative only"
> &lt;jeremie&gt; Which would become a WG NOTE, right?
> &lt;Ian&gt; Typically, yes.
> &lt;Ian&gt; I think there are examples where they are RECS however

This last part shows that the *Informative* is more arbitrary than I thought (unless I misunderstood again). That'd be great if we could come up with clear, written down rules to detect the *Informative* behavior of a document. Without this, we cannot solve this issue.

Please don't hit me if miss the point! I find this very confusing, so I might have gotten lost somewhere...

**Comment by dontcallmedom** [comment]
I don't think IG can publish Recs; at least, that's the first time I hear about such a thing, and I don't think that has ever happened; I haven't dug in the process doc yet to confirm that's forbidden though.

**Comment by dontcallmedom** [comment]
there it is:
"Neither Interest Groups nor Coordination Groups publish Recommendation Track technical reports; "http://www.w3.org/2014/Process-20140801/#GAGeneral

**Comment by draggett** [comment]
> On 16 Apr 2015, at 17:11, Dominique Hazael-Massieux <notifications@github.com> wrote:
> 
> there it is:
> "Neither Interest Groups nor Coordination Groups publish Recommendation Track technical reports; "http://www.w3.org/2014/Process-20140801/#GAGeneral <http://www.w3.org/2014/Process-20140801/#GAGeneral>
> 

I guess we need clarity whether Working Drafts can only be used for REC track technical reports. If so then IG’s should only publish IG Notes, right?

—
   Dave Raggett <dsr@w3.org <mailto:dsr@w3.org>>

**Comment by astorije** [comment]
@dontcallmedom @ianbjacobs then in that case, we have a problem with [Web Payments Use Cases](http://www.w3.org/TR/2015/WD-web-payments-use-cases-20150416/):

> This document was published by the Web Payments Interest Group as a First Public Working Draft. This document is intended to become a W3C Recommendation.

**Comment by dontcallmedom** [comment]
@astorije I agree

**Comment by darobin** [comment]
It's not you who's confused @astorije, it's the way the system works. It makes no sense whatsoever to have informative Recs. If you're just informing, you're not recommending. It is confusing to have WDs leading to Notes because the only thing that distinguishes them in review is a sentence drowned in the middle of the SotD. It also makes no sense to use Notes to signal end of work on something that was initially meant to become a Rec.

What we should have is:

1. Rec track documents, with all of the maturity levels applying *only* to the Rec track.
2. FYI track documents, possibly with things like "Draft Note" but really groups could just keep updating a Note — this is overengineered and then some.
3. An "Abandoned Draft" status (we could even reuse much of the RSCND style) for stuff that's, well, abandoned.

If the people writing the checkers have problems understanding the rules, imagine what it's like for folks just trying to read them.

Open questions (to make this rant actionable):

* The Process allows us to enforce (1) above, it's just a matter of policy. Can we do that, and if so is this under @plehegar's new prerogatives?
* For (2) we could just tell people to keep publishing Notes and never WDs leading to Notes (and I guess that supporting Notes in Echidna would not be hard). Again, same question.
* Item (3) is harder but... I see nothing in the Process that prevents us from creating a new document type (with no IP implications).

**Comment by tripu** [comment]
@plehegar, do you mind reviewing these questions, and maybe summarising what you think it's to be done here&hellip;?

**Comment by tripu** [comment]
@plehegar, any luck reviewing this discussion yourself, or getting someone else to do it&hellip;?

**Comment by deniak** [comment]
We are now extracting that bit from the document as the current pubrules does. Closing this ticket and opening a new one #365 for the open questions.
